lib/fluent/plugin/in_sample.rb

```diff
@@ -117,15 +119,23 @@ def emit(num)
       end
     end
 
+    def sample
+      if @reuse_record
+        @sample[@sample_index]
+      else
+        @sample[@sample_index].dup
+      end
+    end
+
     def generate
-      d = @sample[@sample_index]
+      d = sample
       unless d
         @sample_index = 0
-        d = @sample[@sample_index]
+        d =  sample
       end
       @sample_index += 1
       if @auto_increment_key
-        d = d.dup
+        d = d.dup unless @reuse_record
```

Comment:
Should it be inverted?

```suggestion
        d = d.dup if @reuse_record
```

Reply:
It should be, thanks.